pandas/core/indexes/datetimes.py

```diff
@@ -199,10 +198,11 @@ def _join_i8_wrapper(joinf, **kwargs):
 
     _engine_type = libindex.DatetimeEngine
 
-    tz = None
+    _tz = None
     _freq = None
     _comparables = ['name', 'freqstr', 'tz']
     _attributes = ['name', 'freq', 'tz']
+    timetuple = None
```

Comment:
This is another we'll-need-this-later (when moving from inheritance to composition)

Reply:
The stdlib datetime.datetime comparison methods check if this attribute exists and if so return `NotImplemented`, otherwise raise TypeError for non-datetime objects